include/dali/core/tensor_shape.h

```diff
@@ -244,11 +244,19 @@ struct TensorShape<DynamicDimensions>
    * @brief Return empty shape of specified dimensionality
    */
   static TensorShape<> empty_shape(int dim) {
+    return TensorShape<>::filled_shape(dim, 0);
+  }
+
+  /**
+   * @brief Return shape of specified dimensionality filled with value
+   */
+  static TensorShape<> filled_shape(int dim, int64_t value) {
     assert(dim > 0);
```

Comment:
This assertion isn't necessary for `filled_shape`.
`empty_shape` means "a shape of an empty tensor" - so it needs to have at least one extent equal to zero, so their product is zero (a 0D tensor has a volume of 1, which is consistent with the behavior of NumPy arrays, Torch tensors and the like).
A `filled_shape` doesn't carry any semantics other than the shape contains `value` repeated `dim` times.